pkg/osvscanner/osvscanner.go

```diff
@@ -62,7 +72,7 @@ var OnlyUncalledVulnerabilitiesFoundErr = errors.New("only uncalled vulnerabilit
 //   - Any lockfiles with scanLockfile
 //   - Any SBOM files with scanSBOMFile
 //   - Any git repositories with scanGit
-func scanDir(r reporter.Reporter, query *osv.BatchedQuery, dir string, skipGit bool, recursive bool, useGitIgnore bool) error {
+func scanDir(r reporter.Reporter, dir string, skipGit bool, recursive bool, useGitIgnore bool) ([]Package, error) {
```

Comment:
While i think there is some benefit to using the `Package` type instead of the `osv.BatchedQuery` type here because we're no longer just calling the osv API in this PR, I'm not sure there is much benefit to returning `[]Package` everywhere (rather than updating it in place somehow). To me returning the packages makes it clearer what the function's inputs and outputs are, but I am willing to concede that this is just personal preference.

Reply:
Returning `[]Package` definitely looks better to me as well